### DISCUSSION TITLE
Issue with interconnected checkboxes

### DISCUSSION THREAD

**Discussion by arcontechnologies** [pymupdf/PyMuPDF#1559] [Looking for help]
Hi,

I need to update pdf form based on widgets where I can display / manage widgets :

```
for field in page.widgets():
    if (field.field_name =="Identification du courtier"):
        field.field_value ="BX00600124" 
    

> if (field.field_name =="Civ_1"): 
>         field.field_value = True
> if (field.field_name =="Civ_1"): 
>         field.field_value = False

   
    if (field.field_name =="Nom"): 
        field.field_value ="Dupont-Jacobi"
```
the 2 interconnected checkboxes share the same name "Civ_1". So there's no way to update one without affecting the other. as I understood, such interconnected grouping is not supported for now. is there any workaround to get one checkbox updated without affecting the other?

Thanks for your feeedback.

PS : I'm new to pyMuPDF.

**Comment by JorjMcKie** [comment]
> is there any workaround to get one checkbox updated without affecting the other?

I understand, those buttons are **_RadioButtons_** in the same RBG (radio button group)? If so, then this is not supported by PyMuPDF.

I cannot imagine how otherwise two buttons can be "interconnected" BTW.
If they have just the same name (which shuld actually not happen ...), and there is no other semantic relationship, then try to distinguish between them using their XREF, rectangle or other things.

**Reply by arcontechnologies** [reply]
Thanks for your feedback. I'll give it a try.
But when I'm looking into the type, I'm getting checkbox and not radiobutton :
```
field name : Civ_1 **** type : CheckBox **** button states : {'normal': ['M.'], 'down': ['M.', 'Off']} **** field value : M.   
field name : Civ_1 **** type : CheckBox **** button states : {'normal': ['Mme'], 'down': ['Mme', 'Off']} **** field value : M. 
```

**Comment by ottothorntonsilver-noyo** [comment]
I am experiencing a similar issue where radio buttons are not being recognized as radio buttons. They are returning "CheckBox" when I call `widget.field_type_string`.

**Reply by arcontechnologies** [reply]
Hi sorry for the late answer. I fixed it by using :

```
if (field.field_name =="Civ_1" and field.rect ==(28.350000381469727, 310.8370361328125, 36.96099853515625, 319.447998046875)): 
            field.field_value = True
if (field.field_name =="Civ_1" and field.rect ==(28.350000381469727, 310.8370361328125, 36.96099853515625, 319.447998046875)): 
            field.field_value = True   
```

**field.rect** instead only field name. Actually each object has coordinates in the corresponding pdf. Therefore, you can use it to update you field accordingly. (you can find it in the documentation). I test it and it's pretty accurate and very useful when you need to differentiate fields with the same name.
